sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -27,77 +59,110 @@ impl MetadataFeature {
 
     /// Creates a new [`MetadataFeature`].
     #[inline(always)]
-    pub fn new(data: impl Into<Vec<u8>>) -> Result<Self, Error> {
-        Self::try_from(data.into())
+    pub fn new(data: impl IntoIterator<Item = (Vec<u8>, Vec<u8>)>) -> Result<Self, Error> {
+        let data: BTreeMap<Vec<u8>, Vec<u8>> = data.into_iter().collect();
+
+        for key in data.keys() {
```

Comment:
I'd construct it first and then call `verify_keys_packable` to avoid the code duplication